### ISSUE TITLE
Put strapi in peerDependencies

### ISSUE THREAD

**Issue by jorrit** [pluginpal/strapi-plugin-config-sync#161]
### Describe the bug

`strapi-plugin-config-sync` directly depends on a specific strapi version. This means that when I update strapi to a version different than 5.2.0, I suddenly have two versions installed.

### Steps to reproduce the behavior

1. Install `strapi-plugin-config-sync` and strapi 5.5.1
2. Run `npm ls @strapi/admin`
3. See that multiple strapi versions are installed:

```
> npm ls @strapi/admin
+-- @strapi/strapi@5.5.1
| +-- @strapi/admin@5.5.1
| +-- @strapi/content-manager@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/content-releases@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/content-type-builder@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/core@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/email@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/i18n@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| +-- @strapi/review-workflows@5.5.1
| | `-- @strapi/admin@5.5.1 deduped
| `-- @strapi/upload@5.5.1
|   `-- @strapi/admin@5.5.1 deduped
`-- strapi-plugin-config-sync@3.0.4
  `-- @strapi/admin@5.2.0
```

### Expected behavior

`strapi-plugin-config-sync` should be installable with a range of Strapi versions.

Environment: development
OS: win32-x64
Strapi Version: 5.5.1
Node/Yarn Version: npm/10.9.0 node/v23.3.0 win32 x64 workspaces/false
Edition: Community
Database: postgres

**Comment by boazpoolman** [comment]
Hi @jorrit,

Thank you for reporting this issue.

I've had multiple discussions with @joshuaellis (Strapi employee) about the matter when I was initially migrating this plugin to v5.
I'll try to summarize the conclusions I made during the process.

**Strapi as dependency**
Using `@strapi/sdk-plugin` to bundle your plugin you can do one of two things:

1. Don't put the `@strapi/x` packages you use in your dependencies array.
This will cause those packages to be bundled with your plugin. Causing large bundle sizes and indirectly cause the `@strapi/x` code to be installed anyways.
2. Do put the `@strapi/x` packages you use in your dependencies array.
This will prevent the packages from being bundled with your plugin, but rather be installed alongside them. This is how config-sync is orchestrated and why you've submitted this issue.

**Multiple strapi versions**
Because we've chosen option 2, you may now end up with multiple versions of the same packages in your project. Like `@strapi/admin`. This is by design. It allows Strapi to release breaking changes more frequent, without breaking plugins in the ecosystem.
Source: https://github.com/strapi/strapi/issues/19634#issuecomment-1971942204

I hope this answers your question.

**Comment by jorrit** [comment]
Thanks for your extensive comment, Boaz. I disagree with the statement 'This will cause those packages to be bundled with your plugin. Causing large bundle sizes and indirectly cause the @strapi/x code to be installed anyways.' The plugin should not bundle any other code if peerDependencies are used, it just refers to other plugins and they get installed together on the target system. The dependency range determines which versions are chosen.
Any user of `strapi-plugin-config-sync` is going to have a version of `@strapi/admin` already in their project. If you allow any `5.x` version, no other NPM package is needed when a user installs `strapi-plugin-config-sync`.

It is true that an upgrade of `@strapi/admin` could break the code of `strapi-plugin-config-sync`. I think it is a more pragmatic choice to rely on semver adherance of the strapi maintainers than it is to force any user of `strapi-pugin-config-sync` to install a second copy of `@strapi/admin` and possibly a lot of extra subdependencies.

Anyway, thanks for providing this plugin free of charge, it has been really helpful in developing our sites. I hope you reconsider your stance and try if `peerDependencies` work better.

**Comment by boazpoolman** [comment]
Thanks for your input @jorrit,

Upon further inspection it seems like I may have been wrong about the workings of the `@strapi/sdk-plugin`.
If I put the dependencies in the `peerDependencies` array (which I do agree is where they belong) it also does not bundle them in the plugin.

I will do some further investigation and potentially update the dependency management.

Any updates I will post here.

**Comment by boazpoolman** [comment]
@jorrit I've juggled the dependencies a bit, I believe it should be good now.

Can you do a quick test on your side?
I've published a beta version you can install like so:

```
yarn add strapi-plugin-config-sync@beta

// Or with npm

npm install strapi-plugin-config-sync@beta
```

**Comment by jorrit** [comment]
`> npm install strapi-plugin-config-sync@3.1.0-beta.1 `

`removed 48 packages, changed 1 package, and audited 1815 packages in 5s`

It works, thanks!

**Comment by boazpoolman** [comment]
Great, I've published a stable version with these changes, v3.1.0.
